pkg/k8s/apis/cilium.io/v2alpha1/flowlog.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package v2alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:onlyVerbs=get,list,watch
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium},singular="ciliumflowlog",path="ciliumflowlogs",scope="Cluster",shortName={cfl,ciliumfl}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",description="Time duration since creation of CiliumFlow Logging",name="Age",type=date
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// CiliumFlowLog configures a flow log task for Cilium agents.
+//
+// +deepequal-gen=false
+type CiliumFlowLog struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired specification/configuration of the flow log.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Spec is immutable"
+	Spec FlowLogSpec `json:"spec"`
+
+	// Status defines the realized specification/configuration and status
+	// of the flow log.
+	//
+	// +kubebuilder:validation:Optional
+	Status FlowLogStatus `json:"status,omitempty"`
+}
+
+// FlowLogSpec defines configuration of a flow log task.
+//
+// +deepequal-gen=false
+type FlowLogSpec struct {
+	// List of field names that will be kept in the log output.
+	//
+	// +kubebuilder:validation:Optional
+	FieldMask []string `json:"fieldmask,omitempty"`
+
+	// Filters specifies flows to include in the log output. Flow passes the
+	// filter if any of the entries match. Only Flows matching the filter, but
+	// not matching exclude-filter will be logged. The check is disabled when
+	// empty.
+	//
+	// +kubebuilder:validation:Optional
+	Filters []*flowpb.FlowFilter `json:"filters.omitempty"`
+
+	// ExcludeFilters specifies flows to exclude from the log output. Flow
```

Comment:
"Flow passes the filter if any of the entries match" sounds like we would log the flow.

Maybe "Flows are excluded from logging if any of the exclusion filters match"

Reply:
I also struggle with wording. Do you have other ideas (preferably for both)?